php-checks/src/main/java/org/sonar/php/checks/EmptyMethodCheck.java

```diff
@@ -56,13 +56,25 @@ public void visitMethodDeclaration(MethodDeclarationTree tree) {
 
   @Override
   public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
-    if (!(hasValuableBody(tree.body()) || hasCommentAbove(tree.functionToken()))) {
+    if (!(hasValuableBody(tree.body()) || hasCommentAbove(tree))) {
       commitIssue(tree, "function");
     }
 
     super.visitFunctionDeclaration(tree);
   }
 
+  private static boolean hasCommentAbove(MethodDeclarationTree tree) {
+    if (tree.modifiers().isEmpty()) {
+      return hasCommentAbove(tree.functionToken());
+    } else {
+      return hasCommentAbove(tree.modifiers().get(0));
+    }
```

Comment:
We could also use `PHPTree.getFirstToken()`:
```
return hasCommentAbove(((PHPTree) tree).getFirstToken());
```
`PHPTree` is not in the custom rule API and we have to cast `tree` to `PHPTree`. We can think of defining `getFirstToken` on `Tree` in the near future.